tests/basic/pure_test.go

```diff
@@ -4238,3 +4238,115 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
+				contexts = append(contexts, ScheduleApplicationsOnNamespace(appNamespace, "baseappsdeployment")...)
+			}
+		})
+		stepLog = "Deploy a app which uses FADA volume and validate"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			context, err := Inst().S.Schedule("test-fada-volume", scheduler.ScheduleOptions{
+				AppKeys:            []string{"fio-fa-davol"},
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", "test-fada-volume")
+			contexts = append(contexts, context...)
+
+		})
+		ValidateApplications(contexts)
+
+		stepLog = "Do parallel resize of base apps volume and FADA volume"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			go func() {
+				for _, ctx := range contexts[:1] {
+					var pvcs []*v1.PersistentVolumeClaim
+					appVolumes, err := Inst().S.GetVolumes(ctx)
+					log.FailOnError(err, "Failed to get volumes from context")
+					log.InfoD(fmt.Sprintf("increase volume size %s on app %s's volumes: %v",
+						Inst().V.String(), ctx.App.Key, appVolumes))
+					for _, eachVol := range appVolumes {
+						pvc, err := GetPVCObjFromVol(eachVol)
+						log.FailOnError(err, "Failed to get PVC Details from Volume [%v]", eachVol.Name)
+						pvcs = append(pvcs, pvc)
+					}
+					for _, pvc := range pvcs {
+						pvcSize := pvc.Spec.Resources.Requests.Storage().String()
+						pvcSize = strings.TrimSuffix(pvcSize, "Gi")
+						pvcSizeInt, err := strconv.Atoi(pvcSize)
+						log.InfoD("increasing pvc [%s/%s]  size to %v %v", pvc.Namespace, pvc.Name, 2*pvcSizeInt, pvc.UID)
+						resizedVol, err := Inst().S.ResizePVC(ctx, pvc, uint64(pvcSizeInt))
+						log.FailOnError(err, "pvc resize failed pvc:%v", pvc.UID)
+						log.InfoD("Vol uid %v", resizedVol.ID)
+						resizedVols = append(resizedVols, resizedVol)
+					}
+				}
+			}()
+			for _, ctx := range contexts[1:] {
+				var pvcs []*v1.PersistentVolumeClaim
+				appVolumes, err := Inst().S.GetVolumes(ctx)
+				log.FailOnError(err, "Failed to get volumes from context")
+				log.InfoD(fmt.Sprintf("increase volume size %s on app %s's volumes: %v", Inst().V.String(), ctx.App.Key, appVolumes))
+				for _, eachVol := range appVolumes {
+					pvc, err := GetPVCObjFromVol(eachVol)
+					log.FailOnError(err, "Failed to get PVC Details from Volume [%v]", eachVol.Name)
+					pvcs = append(pvcs, pvc)
+				}
+				for _, pvc := range pvcs {
+					pvcSize := pvc.Spec.Resources.Requests.Storage().String()
+					pvcSizestring := strings.TrimSuffix(pvcSize, "Gi")
+					pvcSizeInt, err := strconv.Atoi(pvcSizestring)
+					log.InfoD("increasing pvc [%s/%s]  size to %v %v", pvc.Namespace, pvc.Name, 2*pvcSizeInt, pvc.UID)
+					resizedVol, err := Inst().S.ResizePVC(ctx, pvc, uint64(pvcSizeInt))
+					log.FailOnError(err, "pvc resize failed pvc:%v", pvc.UID)
+					log.InfoD("Vol uid %v", resizedVol.ID)
+					resizedVols = append(resizedVols, resizedVol)
+				}
+			}
+		})
```

Comment:
I think the code should look more like this. If so, I would suggest making `resizeVolume` return an error instead of throwing it, and please consider using a more meaningful name than `resizeVolume`.

```go
resizeVolume := func(ctx context.Context) {
	var pvcs []*v1.PersistentVolumeClaim
	appVolumes, err := Inst().S.GetVolumes(ctx)
	log.FailOnError(err, "Failed to get volumes from context")
	log.InfoD(fmt.Sprintf("Increase volume size %s on app %s's volumes: %v", Inst().V.String(), ctx.App.Key, appVolumes))

	for _, eachVol := range appVolumes {
		pvc, err := GetPVCObjFromVol(eachVol)
		log.FailOnError(err, "Failed to get PVC Details from Volume [%v]", eachVol.Name)
		pvcs = append(pvcs, pvc)
	}

	for _, pvc := range pvcs {
		pvcSize := pvc.Spec.Resources.Requests.Storage().String()
		pvcSize = strings.TrimSuffix(pvcSize, "Gi")
		pvcSizeInt, err := strconv.Atoi(pvcSize)
		log.FailOnError(err, "Failed to parse PVC size [%v]", pvcSize)
		log.InfoD("Increasing PVC [%s/%s] size to %v %v", pvc.Namespace, pvc.Name, 2*pvcSizeInt, pvc.UID)

		resizedVol, err := Inst().S.ResizePVC(ctx, pvc, uint64(pvcSizeInt*2))
		log.FailOnError(err, "PVC resize failed for PVC:%v", pvc.UID)
		log.InfoD("Resized volume UID: %v", resizedVol.ID)
		resizedVols = append(resizedVols, resizedVol)
	}
}

for _, ctx := range contexts {
	go resizeVolume(ctx)
}
```